File: app/src/main/java/com/song/deviceinfo/ui/battery/BatteryViewModel.java
package com.song.deviceinfo.ui.battery;

import com.song.deviceinfo.info.BatteryInfo;
import com.song.deviceinfo.ui.base.NormalViewModel;

import java.util.List;

import androidx.core.util.Pair;

/**
 * Created by chensongsong on 2020/5/27.
 */
public class BatteryViewModel extends NormalViewModel<Pair<String, String>> {

    @Override
    public List<Pair<String, String>> getNormalInfo() {
        return BatteryInfo.getBatteryInfo();
    }
}

File: app/src/main/java/com/song/deviceinfo/ui/battery/BatteryViewModel.kt
package com.song.deviceinfo.ui.battery

import android.content.Context
import androidx.core.util.Pair
import com.song.deviceinfo.info.BatteryInfo
import com.song.deviceinfo.ui.base.NormalViewModel

/**
 * Created by chensongsong on 2020/5/27.
 */
class BatteryViewModel : NormalViewModel() {
    fun getBatteryInfo(context: Context): List<Pair<String, String>> {
        return BatteryInfo.getBatteryInfo(context)
    }
}